src/telemetry.ts

```diff
@@ -0,0 +1,77 @@
+// This module provides a global interface to the CodeScene telemetry singleton.
+import * as vscode from 'vscode';
+import axios from 'axios';
+import { sign } from './codescene-interop';
+
+
+export default class Telemetry {
+  private static _instance: Telemetry
+
+  private telemetryLogger: vscode.TelemetryLogger;
+
+  constructor(private cliPath: string) {
+    const sender: vscode.TelemetrySender = {
+      sendEventData: async (eventName, eventData)  => {
+        this.postTelemetry(eventName, eventData);
+      },
+      sendErrorData: (error) => {
+        console.log(error);
+      }
+    };
+    this.telemetryLogger = vscode.env.createTelemetryLogger(sender);
+  }
+
+  static init(cliPath: string): void {
+    console.log('CodeScene: initializing telemetry logger');
+    Telemetry._instance = new Telemetry(cliPath);
+  }
+
+  static get instance(): Telemetry {
+    return Telemetry._instance
+  }
+
+  logUsage(eventName: string) {
+    this.telemetryLogger.logUsage(eventName);
+  }
+
+  private postTelemetry(eventName: string, eventData: any) {
+    const config = {
+      headers: { 'content-type': 'application/json' },
+      timeout: 5000 //milliseconds
```

Comment:
I'm not sure what's a good timeout setting but I would be tempted to increase it a bit since people might be running this in various environments with varying network conditions.
Maybe 15-30 seconds.

is there any impact this timeout has on the user using the tool?